dbschema/variance-explanation.esdl

```diff
@@ -0,0 +1,16 @@
+module ProgressReport {
+  type VarianceExplanation {
+    required report: default::ProgressReport {
+      readonly := true;
+    };
+    
+    reasons: array<str> {
+      readonly := true;
+    };
+
+    comments: default::RichText {
+      readonly := true;
```

Comment:
```suggestion
```
Editable

Reply:
The reason we thought these were readonly is because they are marked as such in the dto.

Reply:
That readonly just means the API cannot modify these objects. If it wants to change it, it has to do so with a completely new object identity.
i.e.
```ts
const changed = { ...explanation, reasons: [] };
// disallows
explanation.reasons = [];
```

This is completely independent of whether a value can be changed in DB. For that you should look at the _update_ DTO.